src/selection-recommendation/cost.py

```diff
@@ -0,0 +1,96 @@
+import sys
+sys.path.append("../util")
+import numpy as np
+import Database
+
+DB_NAME = "uoftcourses"
+DB_PATH = "../../database.info"
+
+
+def __get_instructor_scores(instructor, cID):
+    """
+    instructor: a string e.g. "S Huynh", "T Fairgrieve"
+    cID: a string e.g. "CSC148"
+    Returns a list of length 6 with mean evaluation data with a highscore of 5, if evaluation data exists.
+    Otherwise, returns None.
+    """
+    connection = Database.get_connection(DB_PATH, DB_NAME)
+    cursor = connection.cursor()
+
+    all_eval_data = Database.get_eval_data_by_cID_and_instructor(cursor, cID, instructor)
+
+    if len(all_eval_data) == 0:  # no eval data for cID and instructor
+        return None
+
+    intellectually_stimulating = [i[8] for i in all_eval_data]
+    deeper_understanding = [i[9] for i in all_eval_data]
+    course_atmosphere = [i[10] for i in all_eval_data]
+    overall_quality = [i[13] for i in all_eval_data]
+    enthusiasm = [i[14] for i in all_eval_data]
+    recommend = [i[16] for i in all_eval_data]
+    # factors like homework fairness and workload were omitted since they are instructor-independent
+
+    # take the mean across all terms/years
+    means = []
+    means.append(np.mean(intellectually_stimulating))
+    means.append(np.mean(deeper_understanding))
+    means.append(np.mean(course_atmosphere))
+    means.append(np.mean(overall_quality))
+    means.append(np.mean(enthusiasm))
+    means.append(np.mean(recommend))
+
+    return means
+
+
+def all_instructor_scores(schedule):
+    """
+    schedule: a list of lists. This should be an output from a process_schedule() call.
+    e.g. [
+             [None, None, None, None, None, None, None, None, None, None, ('CSC148', 'Lec 5101'), ('CSC148', 'Lec 5101'), None, None],
+             [None, None, None, None, None, None, None, None, None, None, ('CSC165', 'Lec 5101'), ('CSC165', 'Lec 5101'), ('CSC165', 'Lec 5101'), None],
+             [None, None, None, None, None, None, None, None, None, None, ('CSC165', 'Lec 5101'), ('CSC165', 'Lec 5101'), None, None],
+             [None, None, None, None, None, None, None, None, None, None, ('CSC148', 'Lec 5101'), ('CSC148', 'Lec 5101'), ('CSC148', 'Lec 5101'), None],
+             [None, None, None, None, None, None, None, None, None, None, None, None, None, None]
+         ]
+
+    Returns a list of lists, each having length 6. Each list contains the eval data of a particular prof
+    in one course.
+    """
+    connection = Database.get_connection(DB_PATH, DB_NAME)
+    cursor = connection.cursor()
+
+    instructor_scores = []
+    seen = []
```

Comment:
I think it's better to change it to a set, which costs O(1) while list costs O(n)?